testing/upload_validation.ts

```diff
@@ -795,14 +799,50 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     autocomplete: z.unknown().optional(),
     defaultValue: z.unknown().optional(),
     suggestedValue: z.unknown().optional(),
+    fullCrawlSuggestedValue: z.unknown().optional(),
+    allowedPrecannedValues: z.array(z.unknown()).optional(),
+    allowManualInput: z.boolean().optional(),
     crawlStrategy: z.unknown().optional(),
     supportsIncrementalSync: z.boolean().optional(),
-  }).refine(
-    param => {
-      return param.optional || param.supportsIncrementalSync !== false;
-    },
-    {message: 'Required params should support incremental sync.'},
-  );
+  })
+    .refine(
+      param => {
+        return param.optional || param.supportsIncrementalSync !== false;
+      },
+      {message: 'Required params should support incremental sync.'},
+    )
+    .refine(
+      param => {
+        if (!param.allowedPrecannedValues) {
+          return true;
+        }
+        return param.type === Type.date || (isArrayType(param.type) && param.type.items === Type.date);
+      },
+      {message: 'allowedPrecannedValues is not allowed on parameters of this type.'},
+    )
+    .refine(
+      param => {
+        if (!param.allowedPrecannedValues || param.type !== Type.date) {
+          return true;
+        }
+        return param.allowedPrecannedValues?.every(
+          (value: unknown) => typeof value === 'string' && AllRelativeDates.includes(value as PrecannedDate),
+        );
+      },
+      {message: 'allowedPrecannedValues for a date parameter can only be a list of relative dates.'},
+    )
+    .refine(
+      param => {
+        if (!param.allowedPrecannedValues || !(isArrayType(param.type) && param.type.items === Type.date)) {
+          return true;
+        }
+        const relativeDateRanges = Object.values(PrecannedDateRange);
+        return param.allowedPrecannedValues?.every(
+          (value: unknown) => typeof value === 'string' && relativeDateRanges.includes(value as PrecannedDateRange),
+        );
+      },
+      {message: 'allowedPrecannedValues for a date array parameter can only be a list of relative date ranges.'},
```

Comment:
It seems like in the enum definitions, you differentiate "relative" data ranges from other kinds of precanned date ranges, but here it seems like you're describing all date ranges (all of PrecannedDateRange) as "relative".

Reply:
fixed!